packages/itwinui-react/src/core/Popover/Popover.tsx

```diff
@@ -98,32 +107,76 @@ type PopoverInternalProps = {
     animationFrame?: boolean;
     layoutShift?: boolean;
   };
-  /**
-   * By default, the popover will only open on click.
-   * `hover` and `focus` can be manually specified as triggers.
-   */
-  trigger?: Partial<Record<'hover' | 'click' | 'focus', boolean>>;
   role?: 'dialog' | 'menu' | 'listbox';
   /**
    * Whether the popover should match the width of the trigger.
    */
   matchWidth?: boolean;
+} & PopoverInteractionProps &
+  Record<string, any>;
+
+/** If listNavigation is an interaction, it forces the required props for it to be provided */
+type PopoverInteractionConditionalProps =
+  | {
+      interactions?: {
+        listNavigation: true;
+      };
+      interactionsProps: {
+        listNavigation: UseListNavigationProps;
+      };
+    }
+  | {
+      interactions?: {
+        listNavigation?: false;
+      };
+      interactionsProps?: {
+        listNavigation?: UseListNavigationProps;
+      };
+    };
+
+type PopoverInteractionProps = PopoverInteractionConditionalProps & {
+  interactionsProps?: {
+    hover?: UseHoverProps<ReferenceType>;
+  };
 };
```

Comment:
You shouldn't need all these complex types. It would be enough for every interaction to accept an object in addition to a boolean

Usage:
```js
interactions: {
  hover: { move: false },
  click: true,
}
```

Reply:
Why is it so complicated with the merge helper?

Something like this should be enough, I think (untested):

```js
useClick(
  floating.context,
  { enabled: !!interactions.click, ...interactions.click }
),
// …
```

The `!!` will coerce the objects into a boolean; and the `...` will automatically merge the objects (correctly overriding the defaults, while also being a noop if a boolean is used).

Reply:
Since `interactions.click` can also be a `boolean` it gives a type error.

<img width="469" alt="image" src="https://github.com/iTwin/iTwinUI/assets/45748283/54127d15-161a-47c3-945f-7cb76ed9098b">

I can ignore the type error if we're okay with it because like you said, applying spread on a boolean will be no-op in JS.